internal/discovery/foreign.go

```diff
@@ -38,26 +38,34 @@ func (discovery *DiscoveryCtrl) UpdateForeignLAN(data *discoveryData, trustMode
 	}
 }
 
-func (discovery *DiscoveryCtrl) UpdateForeignWAN(data []*TxtData, sd *v1alpha1.SearchDomain) []*v1alpha1.ForeignCluster {
+func (discovery *DiscoveryCtrl) UpdateForeignWAN(data []*AuthData, sd *v1alpha1.SearchDomain) []*v1alpha1.ForeignCluster {
```

Comment:
could you provide a brief comment that describes what this function is intended to do?